### PR TITLE
Expand ETH client.

### PR DESCRIPTION
**Pull request by ggwpez**
In order to reuse our existing code, i changed the ContractInterface
to a ContractBackend, then we can use our Transactor creation functions
and do not need new NewWatchOpts and NewFilterOpts functions.

### CODE REVIEW

File: eth/client.go [hunk 1]
```diff
@@ -37,13 +36,17 @@ type (
 	}
 )
 
+const (
+	defaultPowDepth = uint64(6)
+	gasLimit        = uint64(6000000)
```
Comment (sebastianst) [inline eth/client.go]: There already is a `defaultGasLimit` in line 23. Please use this one and adjust if necessary. It's confusing to have two consts for that.

File: eth/client.go [hunk 2]
```diff
@@ -37,13 +36,17 @@ type (
 	}
 )
 
+const (
+	defaultPowDepth = uint64(6)
```
Comment (sebastianst) [inline eth/client.go]: Please set to 0. You also don't necessarily need to set a const number type, afaik.

File: eth/client.go [hunk 3]
```diff
@@ -108,6 +112,42 @@ func (cl *Client) SubscribeToBlocks() (*BlockSubscription, error) {
 	return &BlockSubscription{sub, blocks}, nil
 }
 
+// SubscribeToDeposited writes past Deposited events and newly received ones
+// into the sink. Can be cancelled via context.
+func (cl *Client) SubscribeToDeposited(ctx context.Context, contract *bindings.Erdstall, epochs []uint64, sink chan *bindings.ErdstallDeposited) error {
```
Comment (sebastianst) [inline eth/client.go]: So `epochs` is optional to only get specific epochs?

File: eth/client.go [hunk 4]
```diff
@@ -169,9 +183,9 @@ func (cl *Client) DeployContracts(params *tee.Parameters) error {
 	ctx, cancel := NewDefaultContext()
 	defer cancel()
 
-	tr, err := cl.NewTransactor(ctx, defaultGasLimit)
+	tr, err := cl.ContractBackend.NewTransactor(ctx, big.NewInt(0), gasLimit, cl.account)
```
Comment (sebastianst) [inline eth/client.go]: use `defaultGasLimit`.

File: eth/client.go [hunk 5]
```diff
@@ -198,6 +212,38 @@ func (cl *Client) DeployContracts(params *tee.Parameters) error {
 	return nil
 }
 
+func (cl *Client) BindContract(ctx context.Context, addr common.Address) (*tee.Parameters, *bindings.Erdstall, error) {
```
Comment (sebastianst) [inline eth/client.go:215]: Nice :sparkles:

**Review by sebastianst — CHANGES_REQUESTED** [review]

File: eth/client.go (hunk 1 above)
Comment (ggwpez) [reply eth/client.go]: Oh fail i did not see that one. Will move the other variable up.

File: eth/client.go (hunk 3 above)
Comment (ggwpez) [reply eth/client.go]: As i understand it, it is not optional. go-ethereum will only return deposited events with the specified epochs. I amended the comment.

Comment (sebastianst) [reply eth/client.go]: I looked it up, it is optional

**Review by sebastianst — APPROVED** [review]

**Pull request merged**